src/ucp/core/ucp_context.c

```diff
@@ -1714,26 +1733,36 @@ void ucp_memory_detect_slowpath(ucp_context_h context, const void *address,
     ucs_status_t status;
     ucp_md_index_t i;
     uct_md_h md;
+    const void *update_address;
+    size_t update_length;
 
     mem_attr.field_mask = UCT_MD_MEM_ATTR_FIELD_MEM_TYPE |
+                          UCT_MD_MEM_ATTR_FIELD_BASE_ADDRESS |
+                          UCT_MD_MEM_ATTR_FIELD_ALLOC_LENGTH |
                           UCT_MD_MEM_ATTR_FIELD_SYS_DEV;
 
     for (i = 0; i < context->num_mem_type_detect_mds; ++i) {
         md     = context->tl_mds[context->mem_type_detect_mds[i]].md;
         status = uct_md_mem_query(md, address, length, &mem_attr);
         if (status == UCS_OK) {
-            mem_info->type    = mem_attr.mem_type;
-            mem_info->sys_dev = mem_attr.sys_dev;
+            mem_info->type         = mem_attr.mem_type;
+            mem_info->sys_dev      = mem_attr.sys_dev;
+            mem_info->base_address = mem_attr.base_address;
+            mem_info->alloc_length = mem_attr.alloc_length;
             if (context->memtype_cache != NULL) {
-                ucs_memtype_cache_update(context->memtype_cache, address,
-                                         length, mem_info);
+                update_address = (mem_attr.alloc_length == -1) ?
```

Comment:
why would it be -1?

Reply:
@yosefe This is a mistake. Will fix this.